lark/lexer.py

```diff
@@ -147,6 +147,8 @@ class Token(str):
     """
     __slots__ = ('type', 'start_pos', 'value', 'line', 'column', 'end_line', 'end_column', 'end_pos')
 
+    __match_args__ = ('type', 'value', 'start_pos', 'line', 'column', 'end_line', 'end_column', 'end_pos')
```

Comment:
I missed this the first time: We shouldn't have everything in `__match_args__`, if you want to match by `start_pos` or `line` for some reason, just use keyword arguments. Allowing everything to be matched via positional arguments would just encourage unmaintainable code. IMO only `type` and `value` should be in `__match_args__`

Reply:
fixed